### PR TITLE
Spec: add expectations starts_with, ends_with

### PR DESCRIPTION
**Pull request by kostya**
i need this many times.

### CODE REVIEW

**Comment by jhass** [comment]
Didn't make my mind up generally but it should be `start_with` and `end_with` respectively, since the proper English is "It should start with foo".

File: src/spec/expectations.cr [hunk 1]
```diff
@@ -200,6 +200,42 @@ module Spec
     end
   end
 
+  # :nodoc:
+  struct StartsWithExpectation(T)
```
Comment (Sija) [inline src/spec/expectations.cr]: `StartsWithExpectation` -> `StartWithExpectation`

File: src/spec/expectations.cr [hunk 2]
```diff
@@ -200,6 +200,42 @@ module Spec
...
+  struct StartWithExpectation(T)
+    def initialize(@expected_value : T)
+    end
+
+    def match(actual_value)
+      actual_value.starts_with?(@expected_value)
+    end
+
+    def failure_message(actual_value)
+      "Expected:   #{actual_value.inspect}\nto start_with: #{@expected_value.inspect}"
```
Comment (Sija) [inline src/spec/expectations.cr]: `start_with` -> `start with`

File: src/spec/expectations.cr [hunk 3]
```diff
@@ -266,6 +302,18 @@ module Spec
       Spec::ContainExpectation.new(expected)
     end
 
+    # Creates an `Expectation` that  passes if actual start_with *expected* (`.starts_with?`).
```
Comment (RX14) [inline src/spec/expectations.cr]: "starts with *expected* (`.starts_with?`)"

File: src/spec/expectations.cr [hunk 4]
```diff
@@ -266,6 +302,18 @@ module Spec
...
+    # Works on `String`.
+    def start_with(expected)
+      Spec::StartWithExpectation.new(expected)
+    end
+
+    # Creates an `Expectation` that  passes if actual end_with *expected* (`.ends_with?`).
```
Comment (RX14) [inline src/spec/expectations.cr]: ditto

**Comment by RX14** [comment]
I'm a bit wary about having a "slippery slope" towards duplicating all collection/string methods as spec expectations, we shouldn't have too many.

**Comment by kostya** [comment]
there is not so many, for strings useful only contain, start_with, end_with and match

**Comment by asterite** [comment]
@kostya By the way, `string_spec.cr` in the std specs could use this new matcher.

**Comment by kostya** [comment]
i not found in string_spec where to replace it.

**Comment by asterite** [comment]
@kostya https://github.com/crystal-lang/crystal/blob/master/spec/std/string_spec.cr#L943-L968

**Comment by straight-shoota** [comment]
@asterite These are specs explicitly for `String#starts_with?`. Using the expectation here would clearly compromise these specs. It hides their true purpose and might even be completely wrong if `StartWithExpectation` wouldn't use `String#starts_with?` internally.

**Comment by asterite** [comment]
Okay... I don't see it as that terrible, but it's also not strictly necessary.

**Comment by asterite** [comment]
So we just need the approval of @RX14

**Pull request merged**